ts/components/ChooserListContainer.tsx

```diff
@@ -0,0 +1,253 @@
+  renderItem: ListRenderItem<T>;
+  onCancel: () => void;
+  isRefreshEnabled: boolean;
+  isRefreshing?: boolean;
+  onRefresh?: () => void;
+  isSearchEnabled: boolean;
+  onSearchItemContainsText?: (item: T, searchText: string) => boolean;
+  noSearchResultsSourceIcon?: ImageSourcePropType;
+  noSearchResultsSubtitle?: string;
+};
+
+type State = {
+  searchText: Option<string>;
+  debouncedSearchText: Option<string>;
+};
+
+const styles = StyleSheet.create({
+  container: {
+    flex: 1
+  },
+  content: {
+    flex: 1
+  },
+  headerNoSearch: {
+    height: customVariables.contentPadding / 2
+  }
+});
+
+/**
+ * A component for view, search and select a list of items
+ * TODO select will be introduced with story https://www.pivotaltracker.com/story/show/167102335
+ */
+export class ChooserListContainer<T> extends React.PureComponent<
+  Props<T>,
+  State
+> {
+  constructor(props: Props<T>) {
+    super(props);
+    this.state = {
+      searchText: this.props.isSearchEnabled ? some("") : none,
+      debouncedSearchText: this.props.isSearchEnabled ? some("") : none
+    };
+  }
+
+  private onPressCancel = () => {
+    this.props.onCancel();
+  };
+
+  private onPressSave = () => {
+    this.onPressCancel();
+  };
+
+  /**
+   * Header SearchBar
+   */
+  private renderSearchBar() {
+    const { searchText } = this.state;
+    return (
+      <React.Fragment>
+        <Body />
+        <Right>
+          {searchText.isSome() ? (
+            <Item>
+              <Input
+                placeholder={I18n.t("global.actions.search")}
+                value={searchText.value}
+                onChangeText={this.onSearchTextChange}
+                autoFocus={true}
+                placeholderTextColor={color(variables.brandGray)
+                  .darken(0.2)
+                  .string()}
+              />
+              <Button onPress={this.onSearchDisable} transparent={true}>
+                <IconFont
+                  name="io-close"
+                  accessible={true}
+                  accessibilityLabel={I18n.t("global.buttons.close")}
+                />
+              </Button>
+            </Item>
+          ) : (
+            <Button onPress={this.handleSearchPress} transparent={true}>
+              <IconFont
+                name="io-search"
+                accessible={true}
+                accessibilityLabel={I18n.t("global.actions.search")}
+              />
+            </Button>
+          )}
+        </Right>
+      </React.Fragment>
+    );
+  }
+
+  /**
+   * Search
+   */
+  private handleSearchPress = () => {
+    this.setState({
+      searchText: some("")
+    });
+  };
+
+  private onSearchTextChange = (text: string) => {
+    this.setState({
+      searchText: some(text)
+    });
+    this.updateDebouncedSearchText(text);
+  };
+
+  private updateDebouncedSearchText = debounce(
+    (text: string) =>
+      this.setState({
+        debouncedSearchText: some(text)
+      }),
+    300
```

Comment:
avoid magic numbers. It could be hard to update it.
Use a const value in place

Reply:
Done in commit 5ba4dc4